### PR TITLE
Simplifying docker contrib

### PR DESCRIPTION
**Pull request by kakoni**
1) Removing docker-build.sh step and building superset container with docker-compose (build command)
2) Removing exposed ports for psql and redis. 
3) Adding new SUPERSET_ENV development that also runs npm watcher
4) Removing npm and superset runserver/worker from docker-init. Dockerfile already builds assets and  starts superset.

### CODE REVIEW

File: contrib/docker/docker-entrypoint.sh [hunk 1]
```diff
@@ -5,6 +5,10 @@ if [ "$#" -ne 0 ]; then
     exec "$@"
 elif [ "$SUPERSET_ENV" = "local" ]; then
     superset runserver -d
+elif [ "$SUPERSET_ENV" = "development" ]; then
+    superset runserver -d &
+    cd /home/work/incubator-superset/superset/assets
```
Comment (brylie) [inline contrib/docker/docker-entrypoint.sh]: Does this directory structure exist inside of the container?

Comment (kakoni) [reply contrib/docker/docker-entrypoint.sh]: Oh yes.  See
https://github.com/apache/incubator-superset/blob/master/contrib/docker/Dockerfile#L11
https://github.com/apache/incubator-superset/blob/master/contrib/docker/Dockerfile#L32-L34

**Review by brylie — APPROVED** [review]

**Comment by ivansabik** [comment]
Would it be worth adding some explicit instructions in the doc of how to run this locally using docker? Since it's not just `docker-compose up`

**Comment by kakoni** [comment]
@ivansabik I thought that it already exists?
https://github.com/apache/incubator-superset/blob/master/docs/installation.rst#user-content-start-with-docker

Oh course if this PR gets merged that documentation has to changed a little.

**Comment by mistercrunch** [comment]
@kakoni mind adjusting the docs as you mentionned as part of this PR?

**Comment by kakoni** [comment]
@mistercrunch I thought I already did?;) https://github.com/apache/incubator-superset/pull/5802/commits/2f67a04679de80fe2aead114ac06f78169f71485

**Comment by victornoel** [comment]
@kakoni this is potentially superseded by this PR: #5966.
Could you take a look and see if it does indeed? I am sorry I didn't see your PR before. Anyway mine goes farther but seems to tackle what you did.

**Comment by kakoni** [comment]
@victornoel Hi! We seem to have same goals, so we should combine our work.
Anyway question; Build processes/docker images for develoment / production would look bit different (for instance running editable pip install, npm/webpack dev server running for assets). So perhaps, would it make sense to create two compose files(one for dev and one for prod like builds) ?

**Comment by victornoel** [comment]
@kakoni from what I understood, nobody uses this image for production, most people are using https://hub.docker.com/r/amancevice/superset/, so for now, I wouldn't bother tackling this question.

In #5966, the objective is clearly to simplify the experience of the developer, both for the backend and the frontend.
Some things that differs in your PR:
- your build the frontend in the entrypoint.sh. My take was to build it once in the image, and if the desire was to do development, the dev would run dev-server on its host, not in docker (that's not needed to do it in docker since running dev-server is happening all in memory). That being said, it means that if the frontend code change but the dev does not want to run dev-server, then he has to rebuild the image… that's maybe not the best, we could include building the front as part of starting the image.
- yours build the image directly from docker-compose while mine lets the dev build it himself. I think you solution is more elegant.

WDYT? Would you agree that I incorporate those changes in my PR and that we continue with it? Since it covers more things too.

**Comment by brylie** [comment]
Also, is there an official docker image published on DockerHub? That would be helpful, so I can ask our DevOps architect how to set up a Superset instance in our AWS infrastructure.

**Comment by victornoel** [comment]
@brylie you should open another issue for this, I don't think this is the best place to discuss it, @kakoni and I have no official position in the superset org :)
You can check out https://hub.docker.com/r/amancevice/superset/ also.

**Comment by kakoni** [comment]
@victornoel 
>Would you agree that I incorporate those changes in my PR and that we continue with it? Since it covers more things too.

Makes sense. Lets continue there.

**Pull request closed without merging**